### PR TITLE
fix: enable and fixes rules from testifylint on test package

### PR DESCRIPTION
**Pull request by mmorel-35**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This fixes rules from testifylint on test package
They will be enabled on golangci once fixed for all other packages

### CODE REVIEW

File: test/e2e/framework/pod/utils_test.go [hunk 1]
```diff
@@ -65,7 +66,7 @@ func TestMixinRestrictedPodSecurity(t *testing.T) {
 	for _, pod := range restrictablePods {
 		t.Run(pod.Name, func(t *testing.T) {
 			p := pod // closure
-			assert.NoError(t, MixinRestrictedPodSecurity(&p))
+			require.NoError(t, MixinRestrictedPodSecurity(&p))
 			assert.Equal(t, GetRestrictedPodSecurityContext(), p.Spec.SecurityContext,
 				"Mixed in PodSecurityContext should equal the from-scratch PodSecurityContext")
 			assert.Equal(t, GetRestrictedContainerSecurityContext(), p.Spec.Containers[0].SecurityContext,
```
Comment (neolit123) [inline test/e2e/framework/pod/utils_test.go:72]: what is the issue with using assert.NoError?
the remaining of the assertion is done with the assert package?

**Review by neolit123 — COMMENTED** [review]
/cc @pohly

File: test/e2e/framework/pod/utils_test.go (hunk 1 above)
Comment (neolit123) [reply test/e2e/framework/pod/utils_test.go:72]: that seems like some testify weirdness unless the whole assert package is a soft-assert. i would have assumed that assert.Error and NoError block further execution.

Comment (neolit123) [reply test/e2e/framework/pod/utils_test.go:72]: it seems like so:

https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1579

https://github.com/stretchr/testify/blob/master/assert/assertions.go#L348

it calls t.Errorf

**Comment by neolit123** [comment]
/lgtm
/ok-to-test

i think we should settle on goomega vs testify for all code in /test. using testify for ginkgo tests is a bit odd.

**Comment by neolit123** [comment]
/retest
unrelated flake
https://github.com/kubernetes/kubernetes/issues/125760

**Comment by pohly** [comment]
> i think we should settle on goomega vs testify for all code in /test. using testify for ginkgo tests is a bit odd.

It's perhaps unfamiliar, but it is possible to use testify with Ginkgo and Gomega with `go test`. We already use both in the project, so why should we restrict a developer to picking one over the other? We lost the "consistency" already a long time ago when picking (or favoring) Gomega for E2E and testify for `go test`.

Speaking of `/test`, isn't that used in both, at least in theory?

I'd love to make everything usable in both unit/integration tests and E2E tests - see https://pkg.go.dev/k8s.io/kubernetes@v1.30.2/test/utils/ktesting. Currently it is for `go test` tests, https://github.com/kubernetes/kubernetes/pull/122481 adds support for E2E. But it looks like I first need to explain this better and figure out whether it has the support from others before I can proceed...

**Pull request merged**